addons/stock/models/stock_picking.py

```diff
@@ -559,6 +559,8 @@ def do_print_picking(self):
 
     @api.multi
     def action_confirm(self):
+        if not all(self.env['res.users'].browse(self._uid).company_id != p.company_id for p in self):
```

Comment:
`if not all(self.env.user.company_id != p.company_id for p in self):`

Reply:
@fmdl done